x/ethbridge/keeper/keeper.go

```diff
@@ -57,18 +62,19 @@ func (k Keeper) ProcessClaim(ctx sdk.Context, claim *types.EthBridgeClaim) (orac
 // ProcessSuccessfulClaim processes a claim that has just completed successfully with consensus
 func (k Keeper) ProcessSuccessfulClaim(ctx sdk.Context, claim *types.EthBridgeClaim) error {
 	logger := k.Logger(ctx)
+	tokenMetadata, ok := k.GetTokenMetadata(ctx, claim.DenomHash)
+	if !ok {
+		return fmt.Errorf("token metadata not available for %s", claim.DenomHash)
+	}
 
 	var coins sdk.Coins
 	var err error
 	switch claim.ClaimType {
 	case types.ClaimType_CLAIM_TYPE_LOCK:
-		symbol := fmt.Sprintf("%v%v", types.PeggedCoinPrefix, claim.Symbol)
-		k.AddPeggyToken(ctx, symbol)
-
-		coins = sdk.Coins{sdk.NewCoin(symbol, claim.Amount)}
+		coins = sdk.Coins{sdk.NewCoin(tokenMetadata.Symbol, claim.Amount)}
 		err = k.bankKeeper.MintCoins(ctx, types.ModuleName, coins)
 	case types.ClaimType_CLAIM_TYPE_BURN:
-		coins = sdk.Coins{sdk.NewCoin(claim.Symbol, claim.Amount)}
+		coins = sdk.Coins{sdk.NewCoin(tokenMetadata.Symbol, claim.Amount)}
 		err = k.bankKeeper.MintCoins(ctx, types.ModuleName, coins)
```

Comment:
Here we should transfer coins from the ethbridge module to the user instead of minting them coins.

Reply:
Agree, no need to mint for the burn case.